### ISSUE TITLE
Initial release

### ISSUE THREAD

**Issue by cswinter** [entity-neural-network/incubator#192]
# Initial Release

Entity Gym and ENN-PPO already have sufficiently many novel features to be useful. We should aim to tie off some loose ends and release and publicize an initial version of the library. The tentative release date is ~April 29~ ~May 13~ May 16.

## Remaining Tasks

Tasks which need to be completed before release:

- [x] Improve documentation (DRI: @cswinter)
- [x] Clean up public API (DRI: @cswinter)
- [x] https://github.com/entity-neural-network/incubator/issues/24 (DRI: @cswinter)
- [x] https://github.com/entity-neural-network/incubator/issues/75
- [x] Simpler interface for single-agent environments
- [x] Simpler API for dynamic observation/action spaces
- [x] Well-tuned default hyperparameters for enn-ppo (DRI: @cswinter)
- [x] GitHub action to publish enn-ppo and entity-gym packages to PyPI (DRI: @cswinter)

## Stretch Goals

Various additional features that would be nice to have, but aren't essential:

- [ ] entity-neural-network/entity-gym#4 (DRI: @Bam4d)
- [ ] entity-neural-network/entity-gym#2
- [x] https://github.com/entity-neural-network/incubator/issues/26
- [ ] entity-neural-network/entity-gym#1
- [x] https://github.com/entity-neural-network/incubator/issues/190 (DRI: @cswinter)
- [ ] entity-neural-network/enn-trainer#1
- [x] https://github.com/entity-neural-network/incubator/issues/161 (DRI: @cswinter)
- [ ] Rigorous comparison of sample/compute efficiency between RogueNet/enn-ppo and SOTA for vision based policy on a nontrivial environment
